python/ray/data/_internal/execution/operators/task_pool_map_operator.py

```diff
@@ -150,3 +151,20 @@ def incremental_resource_usage(self) -> ExecutionResources:
 
     def get_concurrency(self) -> Optional[int]:
         return self._concurrency
+
+    def all_inputs_done(self):
+        super().all_inputs_done()
+
+        if (
+            self._concurrency is not None
+            and self._metrics.num_inputs_received < self._concurrency
+        ):
```

Comment:
for the task-pool map, this concurrency is the max concurrency, not the min. 
So I think it's fine not to emit the warning

Reply:
Discussed offline -- this warning is useful in the case where you configure `concurrency` with the expectation of that exact concurrency, even though that isn't the actual semantic.

If users find this warning unhelpful or confusing, we can revisit and remove.